src/MSRIOGroup.cpp

```diff
@@ -280,11 +284,160 @@ namespace geopm
                                                    IOGroup::M_UNITS_WATTS,
                                                    agg_function(msr_name),
                                                    ps.description + "\n    alias_for: " + ps.msr_name + " rate of change",
-                                                   IOGroup::M_SIGNAL_BEHAVIOR_VARIABLE};
+                                                   IOGroup::M_SIGNAL_BEHAVIOR_VARIABLE,
+                                                   string_format_double};
             }
         }
     }
 
+    void MSRIOGroup::register_pcnt_scalability_signals(void)
+    {
+        // register time signal; domain board
+        std::string time_name = "MSR::TIME";
+        std::shared_ptr<Signal> time_sig = std::make_shared<TimeSignal>(m_time_zero, m_time_batch);
+        m_signal_available[time_name] = {std::vector<std::shared_ptr<Signal> >({time_sig}),
+                                         GEOPM_DOMAIN_CPU,
+                                         IOGroup::M_UNITS_SECONDS,
+                                         Agg::select_first,
+                                         "Time in seconds used to calculate power"};
+        int derivative_window = 8;
+        double sleep_time = 0.005;  // 5000 us
+
+        // Mapping of high-level signal name to description and
+        // underlying energy MSR.  The domain will match that of the
+        // energy signal.
+        struct cnt_data
+        {
+            std::string cnt_name;
+            std::string description;
+            std::string msr_name;
+        };
+        std::vector<cnt_data> cnt_signals {
+            {"MSR::PPERF:PCNT_RATE",
+                    "Average cpu pcnt value over 20 ms or 4 control loop iterations",
```

Comment:
How can you make any assertions about the control loop cadence here?  Isn't that agent specific?